skrub/_utils.py

```diff
@@ -156,22 +159,85 @@ def repr_args(args, kwargs, defaults={}):
     )
 
 
-def transformer_output_type_error(transformer, transform_input, transform_output):
-    module = sbd.dataframe_module_name(transform_input)
+def set_output(transformer, X):
+    if not hasattr(transformer, "set_output"):
+        return
+    module_name = sbd.dataframe_module_name(X)
+    if module_name == "polars" and parse_version(sklearn.__version__) < parse_version(
+        "1.4"
+    ):
+        # TODO: remove when scikit-learn 1.3 support is dropped.
+        warnings.warn(
+            "To use polars, we recommend using scikit-learn version 1.4 or newer. The"
+            f" current version is {sklearn.__version__!r}. This will produce correct"
+            " results but may cause an extra copy of the transformer's output."
+        )
+        transformer.set_output(transform="pandas")
+        return
```

Comment:
This is not cover but I assume that it will be in my PR.

Reply:
yes exactly -- that's how we found the polars and scikit-learn 1.3 issue in the first place. I thought it's not worth adding a dedicated workflow in the CI here because it will be superseded by your PR anyway